soapui/src/test/java/com/eviware/soapui/impl/wsdl/submit/filters/HttpAuthenticationRequestFilterTest.java

```diff
@@ -47,13 +57,42 @@ public void setup() {
     }
 
     @Test
-    public void selectingAuthTypeNtlmReturnsNtlmEvenIfSpnegoRequested() {
+    public void usingNtlmWhenBothNtlmAndNegotiateRequested() {
         selectAuthMethod(CredentialsConfig.AuthType.NTLM);
 
-        filter.filterAbstractHttpRequest(null, wsdlRequest);
+        CredentialsProvider credentialsProvider = makeCredentialsProvider();
+        AuthScope authScope = Mockito.mock(AuthScope.class);
 
-        AuthScheme scheme = getSchemeFor(AuthPolicy.SPNEGO);
-        assertThat(scheme, instanceOf(NTLMScheme.class));
+        when(authScope.getScheme()).thenReturn(AuthPolicy.SPNEGO);
+        assertNull(credentialsProvider.getCredentials(authScope));
+
+        when(authScope.getScheme()).thenReturn(AuthPolicy.NTLM);
+        assertThat(credentialsProvider.getCredentials(authScope), instanceOf(NTCredentials.class));
+    }
+
+    @Test
+    public void usingNegotiateWhenBothNtlmAndNegotiateRequested() {
+        selectAuthMethod(CredentialsConfig.AuthType.SPNEGO_KERBEROS);
+
+        CredentialsProvider credentialsProvider = makeCredentialsProvider();
+        AuthScope authScope = Mockito.mock(AuthScope.class);
+
+        when(authScope.getScheme()).thenReturn(AuthPolicy.NTLM);
+        assertNull(credentialsProvider.getCredentials(authScope));
+
+        when(authScope.getScheme()).thenReturn(AuthPolicy.SPNEGO);
```

Comment:
I believe class AuthPolicy is deprecated. If you have time, replace it with org.apache.http.client.config.AuthSchemes.